Library/Homebrew/utils/analytics.rb

```diff
@@ -8,7 +8,7 @@ def analytics_label
 end
 
 def report_analytics(type, metadata = {})
-  return if ENV["HOMEBREW_NO_ANALYTICS"]
+  return if ENV["HOMEBREW_NO_ANALYTICS"] || ENV["HOMEBREW_NO_ANALYTICS_THIS_RUN"]
```

Comment:
I would think so, but I'm being extra defensive here, just in case a code path some how got to this point without going through the normal `setup-analytics`. Which it shouldn't, if it was run in a supported manner, but I don't want to rely on that caveat here. Want to make sure a user-supplied `HOMEBREW_NO_ANALYTICS` is respected no matter what.